pennylane/ftqc/quantum_script_sequence.py

```diff
@@ -0,0 +1,284 @@
+    Intermediate tapes may return mid-circuit measurements, or nothing. This is not currently
+    validated, but it is assumed. The final tape returns terminal measurements."""
+
+    def __init__(self, tapes, shots=None):
+
+        if shots is None:
+            shots = [tape.shots for tape in tapes]
+            if len(set(shots)) != 1:
+                raise RuntimeError(
+                    "All scripts in a QuantumScriptSequence must have the same shots"
+                )
+            shots = shots[0]
+        else:
+            if not isinstance(shots, Shots):
+                shots = Shots(shots)
+        self._shots = shots
+
+        self._tapes = []
+
+        for tape in tapes:
+            aux_tape = QuantumScript(
+                tape.operations,
+                tape.measurements,
+                shots=[1],
+            )
+            self._tapes.append(aux_tape)
+
+        self._index = 0
+
+    @property
+    def tapes(self):
+        """Returns all the tapes in the sequence.
+
+        Returns:
+            list[QuantumScript]: list of all tapes
+        """
+        return self._tapes
+
+    @property
+    def final_tape(self):
+        """Returns the final tape in the sequence.
+
+        Returns:
+            QuantumScript: the final tape in the sequence, with terminal measurements
+        """
+        return self._tapes[-1]
+
+    @property
+    def intermediate_tapes(self):
+        """Returns all but the final tape in the sequence.
+
+        Returns:
+            list[QuantumScript]: list of all tapes except the tape with terminal measurements
+        """
+
+        return self._tapes[:-1]
+
+    @property
+    def measurements(self):
+        """Returns the final measurements for the sequence.
+
+        Returns:
+            list[.MeasurementProcess]: list of measurement processes
+        """
+        return self.final_tape.measurements
+
+    @property
+    def intermediate_measurements(self):
+        """Returns the intermediate measurements for all but the final tape. Since these
+        are in the middle of an execution, they are expected to be empty, or to be mid-circuit
+        measurements.
+
+        Returns:
+            list[list[MidMeasureMP]]: nested list of the returned MCMs for all but the final tape
+        """
+        return [tape.measurements for tape in self.intermediate_tapes]
+
+    @property
+    def operations(self):
+        """Returns the operations for each tape
+
+        Returns:
+            list[list[Operation]]: a nested list of the operations for each tape
+        """
+        return [tape.operations for tape in self.tapes]
+
+    @cached_property
+    def wires(self):
+        """Returns the wires used in the quantum script process
+
+        Returns:
+            ~.Wires: wires in quantum script process
+        """
+        wires = self.tapes[0].wires
+        for tape in self.tapes[1:]:
+            wires += tape.wires
```

Comment:
Nothing to be done here now, but I wonder if we can do this without the implicit update per loop step. I guess it doesn't matter for our current use cases as we should hit scaling issues here.